src/imp/windows.rs

```diff
@@ -0,0 +1,166 @@
+// Distributed under the OSI-approved BSD 3-Clause License.
+// See accompanying LICENSE file for details.
+
+// Implemented by JR Andreassen jr at prtgconsultants com
+// references: 
+//   https://github.com/denisbrodbeck/machineid 
+//   https://github.com/jojonv/rust-win32error/blob/master/src/lib.rs
+
+use winreg::enums::{
+    HKEY_LOCAL_MACHINE, KEY_READ,
+};
+use log::debug;
+use thiserror::Error;
+use uuid::{self, Uuid};
+use std::io::{self};
+
+use rust_win32error::Win32Error;
+
+#[derive(Debug, Error)]
+enum Error {
+    #[error("I/O error")]
+    Io {
+        #[from]
+        source: io::Error,
+    },
+    #[error("W32 error")]
+    W32Err {
+        #[from]
+        source: rust_win32error::Win32Error,
+    },
+    #[error("parse error")]
+    Parse {
+        #[from]
+        source: uuid::Error,
+    },
+}
+
+
+const WIN_REG_BASE_KEY   :&'static str = r"SOFTWARE\\Microsoft\\Cryptography";
```

Comment:
The `r""` should mean that you just need a single `\` right?

Reply:
I believe that is correct